src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs

```diff
@@ -171,19 +171,26 @@ public RunspaceDetails CurrentRunspace
             ProfilePaths profilePaths,
             HashSet<string> featureFlags,
             bool enableConsoleRepl,
+            bool useLegacyReadLine,
             PSHost internalHost,
             HostDetails hostDetails,
             string[] additionalModules
             )
         {
             var logger = factory.CreateLogger<PowerShellContextService>();
 
-            // PSReadLine can only be used when -EnableConsoleRepl is specified otherwise
-            // issues arise when redirecting stdio.
+            // We should only use PSReadLine if we specificied that we want a console repl
+            // and we have not explicitly said to use the legacy ReadLine.
+            // We also want it if we are either:
+            // * On Windows on any version OR
+            // * On Linux or macOS on any version greater than or equal to 7
+            var shouldUsePSReadLine = enableConsoleRepl && !useLegacyReadLine
+                && (VersionUtils.IsWindows || (!VersionUtils.IsWindows && !VersionUtils.IsPS6));
```

Comment:
Shouldn't `(IsWindows || (!IsWindows && IsPS6)` be the same as `(IsWindows || !IsPS6)`?

Reply:
Good call - fixed!